Taarafo.Portal.Web/Services/Foundations/Comments/CommentService.Expections.cs

```diff
@@ -0,0 +1,77 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using RESTFulSense.Exceptions;
+using Taarafo.Portal.Web.Models.Comments;
+using Taarafo.Portal.Web.Models.Comments.Exceptions;
+using Xeptions;
+
+namespace Taarafo.Portal.Web.Services.Foundations.Comments
+{
+    public partial class CommentService
+    {
+        private delegate ValueTask<Comment> ReturningCommentFunction();
+
+        private async ValueTask<Comment> TryCatch(ReturningCommentFunction returningCommentFunction)
+        {
+            try
+            {
+                return await returningCommentFunction();
+            }
+            catch (NullCommentException nullCommentException)
+            {
+                throw CreateAndLogValidationException(nullCommentException);
+            }
+            catch (InvalidCommentException invalidCommentException)
+            {
+                throw CreateAndLogValidationException(invalidCommentException);
+            }
+            catch (HttpRequestException httpRequestException)
+            {
+                var failedCommentDependencyException =
+                    new FailedCommentDependencyException(httpRequestException);
+
+                throw CreateAndLogCriticalDependencyException(failedCommentDependencyException);
+            }
+            catch (HttpResponseUrlNotFoundException httpResponseUrlNotFoundException)
+            {
+                var failedCommentDependencyException =
+                    new FailedCommentDependencyException(httpResponseUrlNotFoundException);
+
+                throw CreateAndLogCriticalDependencyException(failedCommentDependencyException);
+            }
+            catch (HttpResponseUnauthorizedException httpResponseUnauthorizedException)
+            {
+                var failedCommentDependencyException =
+                    new FailedCommentDependencyException(httpResponseUnauthorizedException);
+
+                throw CreateAndLogCriticalDependencyException(failedCommentDependencyException);
+            }
```

Comment:
@cjdutoit there are missing exceptions here.
Bad Request, Conflict, HttResponse ... etc etc.

Reply:
@hassanhabib, have added the missing ones.